File: blog/src/main/java/com/yky/blog/common/entity/Menu.java
package com.yky.blog.common.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("menu")
public class Menu extends BaseEntity {

    @TableId(type = IdType.AUTO)
    private Long id;
    private Long parentId;
    private String title;
    private String type;
    private String path;
    private String component;
    private String perm;
    private String icon;
    private Integer sort;
    private Integer hidden;
}

File: blog-admin/src/api/menu.ts
import request from '@/utils/request'

export interface MenuNode {
  id: number
  parentId: number
  title: string
  type: string
  path?: string
  component?: string
  perm?: string
  icon?: string
  sort: number
  hidden: number
  children?: MenuNode[]
}

export interface MenuSaveDTO {
  parentId?: number
  title: string
  type: string
  path?: string
  component?: string
  perm?: string
  icon?: string
  sort?: number
  hidden?: number
}

export const getMenuTree = () => request.get('/admin/menu/tree')

export const saveMenu = (data: MenuSaveDTO) => request.post('/admin/menu', data)

export const updateMenu = (id: number, data: MenuSaveDTO) => request.put(`/admin/menu/${id}`, data)

export const deleteMenu = (id: number) => request.delete(`/admin/menu/${id}`)
